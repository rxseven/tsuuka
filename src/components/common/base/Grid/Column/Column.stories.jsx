import React from 'react';
import { storiesOf } from '@storybook/react';

import mock from 'stories/mock';
import notes from 'stories/shared/notes';
import Column from './index';

const { content } = mock.elements;
const title = 'Components/Common/Base/Grid/Column';
const component = <Column>{content}</Column>;

storiesOf(title, module).add('default', () => component, {
  notes: notes.minimal
});
