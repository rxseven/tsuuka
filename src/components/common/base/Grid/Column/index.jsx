import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';

const propTypes = exact({
  children: PropTypes.node.isRequired,
  size: PropTypes.string
});

const defaultProps = {
  size: 'col'
};

function Column({ children, size }) {
  return <div className={size}>{children}</div>;
}

Column.propTypes = propTypes;
Column.defaultProps = defaultProps;

export default Column;
