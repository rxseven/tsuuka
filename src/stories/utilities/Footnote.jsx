import React from 'react';
import styled from 'styled-components';

const Frame = styled.div`
  bottom: 2rem;
  font-size: 14px;
  position: fixed;
`;

function Footnote({ children }) {
  return <Frame>{children}</Frame>;
}

export default Footnote;
