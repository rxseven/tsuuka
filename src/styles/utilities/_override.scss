// Reset button's default styles
button:focus {
  box-shadow: none !important;
  outline: 0 !important;
}
