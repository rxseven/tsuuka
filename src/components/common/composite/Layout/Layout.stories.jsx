import React from 'react';
import { storiesOf } from '@storybook/react';

import Grid from 'components/common/base/Grid';
import mock from 'stories/mock';
import notes from 'stories/shared/notes';
import Layout from './index';

const { text } = mock;
const title = 'Components/Common/Composite/Layout';
const props = { children: text };

storiesOf(title, module)
  .addDecorator(story => <Grid.Container>{story()}</Grid.Container>)
  .add('default', () => <Layout {...props} />, {
    notes: notes.minimal
  })
  .add('alignment & size', () => (
    <Layout {...props} alignment="justify-content-end" size="col-sm-6" />
  ));
