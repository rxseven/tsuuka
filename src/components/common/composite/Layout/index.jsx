import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';

import Grid from 'components/common/base/Grid';

const propTypes = exact({
  adaptive: PropTypes.bool,
  alignment: PropTypes.string,
  children: PropTypes.node.isRequired,
  size: PropTypes.string
});

const defaultProps = {
  adaptive: false,
  alignment: 'justify-content-sm-center',
  size: 'col'
};

function Layout({ adaptive, alignment, children, size }) {
  return (
    <Grid.Row alignment={alignment}>
      <Grid.Column size={adaptive ? 'col-md-10 col-lg-8' : size}>
        {children}
      </Grid.Column>
    </Grid.Row>
  );
}

Layout.propTypes = propTypes;
Layout.defaultProps = defaultProps;

export default Layout;
