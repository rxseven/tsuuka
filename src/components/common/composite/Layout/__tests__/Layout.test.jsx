import React from 'react';

import Column from 'components/common/base/Grid/Column';
import { factory, toMarkup } from 'tests/utilities';
import Layout from '../index';

// Mock
jest.mock('components/common/base/Grid/Column', () =>
  jest.fn(({ children }) => <div>{children}</div>)
);

// Arrange
const context = expect.any(Object);
const seed = { content: 'content' };
const source = { children: <span>{seed.content}</span> };
const input = { ...seed, ...source };

// Setup
function setup(props) {
  return factory(Layout, source, props);
}

// Test suites
describe('<Layout />', () => {
  it('should render without crashing', () => {
    setup();
  });

  it('should render passed children correctly', () => {
    const expected = {
      content: input.content,
      html: input.children
    };
    const { component } = setup();

    expect(component).toHaveTextContent(expected.content);
    expect(component).toContainHTML(toMarkup(expected.html));
  });

  it('should align content at the center of the container by default', () => {
    const expected = { class: 'justify-content-sm-center' };
    const { component } = setup();

    expect(component).toHaveClass(expected.class);
  });

  it('should align content to the left of the container', () => {
    const props = { alignment: 'justify-content-sm-left' };
    const expected = { class: props.alignment };
    const { component } = setup(props);

    expect(component).toHaveClass(expected.class);
  });

  it('should render content within a single grid column by default', () => {
    const expected = {
      arguments: {
        children: input.children,
        size: 'col'
      }
    };
    setup();

    expect(Column).toHaveBeenCalledWith(expected.arguments, context);
  });

  it('should render content within six grid columns', () => {
    const props = { size: 'col-md-6' };
    const expected = {
      arguments: {
        children: input.children,
        size: props.size
      }
    };
    setup(props);

    expect(Column).toHaveBeenCalledWith(expected.arguments, context);
  });

  it('should render content within an "adaptive" layout', () => {
    const props = { adaptive: true };
    const expected = {
      arguments: {
        children: input.children,
        size: 'col-md-10 col-lg-8'
      }
    };
    setup(props);

    expect(Column).toHaveBeenCalledWith(expected.arguments, context);
  });
});
