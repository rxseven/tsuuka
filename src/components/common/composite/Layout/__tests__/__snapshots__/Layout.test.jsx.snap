// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> Snapshot tests should render correctly 1`] = `
<div
  className="row justify-content-sm-center"
>
  <div
    className="col"
  >
    content
  </div>
</div>
`;
